payment-processor/src/main/kotlin/org/ostelco/prime/paymentprocessor/StripePaymentProcessor.kt

```diff
@@ -204,52 +204,61 @@ class StripePaymentProcessor : PaymentProcessor {
 
     override fun authorizeCharge(customerId: String, sourceId: String?, amount: Int, currency: String): Either<PaymentError, String> {
         val errorMessage = "Failed to authorize the charge for customerId $customerId sourceId $sourceId amount $amount currency $currency"
-        return either(errorMessage) {
-            val chargeParams = mutableMapOf(
-                    "amount" to amount,
-                    "currency" to currency,
-                    "customer" to customerId,
-                    "capture" to false)
-            if (sourceId != null) {
-                chargeParams["source"] = sourceId
+        return when {
+            amount == 0 -> Either.right("ZERO_CHARGE_IS_OK")
```

Comment:
I think we should  make the chargeId unique (may be add a uuid at  the end e.g. ZERO_CHARGE_IS_OK_<UUID>)